File: Ejercicios/Ejercicio2.py
#  2. Create a currency converter between CLP, ARS, USD, EUR, TRY, GBP with the following features:
# * 		The user must choose their initial currency and the currency they want to exchange to.
# * 		The user can choose whether or not to withdraw their funds. If they choose not to withdraw, it should return to the main menu.
# * 		If the user decides to withdraw the funds, the system will charge a 1% commission.
# * 		Set a minimum and maximum amount for each currency, it can be of your choice.
# * 		The system should ask the user if they want to perform another operation. If they choose to do so, it should restart the process; otherwise, the system should close.
import sys

class menu:

    def __init__(self) -> None:
        self.select()
  

    def select(self):
        while True:
            print("----------------------")
            print("1. view coins")
            print("2. change")
            print("3. exit")
            print("----------------------")
            opcion = int(input(f"Enter option: "))
            self.algo(opcion)
            
        
    
    def algo(self, opcion):
        if opcion == 1:
            values()
        elif opcion == 2:
            change()
        elif opcion == 3:
            sys.exit()
        else:
            print("Invalid option")

class values(menu):

    def __init__(self) -> None:
        self.coins()
        self.view_coins()


    def coins(self):

        self.clp = 888
        self.arg = 811
        self.eur = 0.91
        self.lira = 29
        self.gbp = 0.79
    
    def view_coins(self):
        print("----------------------")
        print("Values \n 1 dollar equals :")
        print(f" clp = {self.clp}\n arg = {self.arg} \n eur = {self.eur}\n try = {self.lira} \n gdp = {self.gbp}")
        print("----------------------")


class change(values):
    

    def __init__(self) -> None:
        super().__init__()
        initial_currency = input("choose a coin between clp, arg, eur, try, gdp: ")
        print(f"enter amount of $ {initial_currency}")

        if initial_currency == "clp":
            initial_currency = self.clp
        elif initial_currency == "arg":
            initial_currency = self.arg
        elif initial_currency == "eur":
            initial_currency = self.eur
        elif initial_currency == "try":
            initial_currency = self.lira
        elif initial_currency == "gbp":
            initial_currency = self.gbp
        else:
            print("Wrong coin")
            print("Closing program...")
            sys.exit()
        print (initial_currency)
        
        
        amount = float(input("ingresar cantidad a cambiar: "))
        total_dollars = amount / initial_currency

        final_coin = input("Enter the coin you want to convert to:")
        print(f"coin to convert: {final_coin}")
        if final_coin == "clp":
            final_coin = self.clp
        elif final_coin == "arg":
            final_coin = self.arg
        elif final_coin == "eur":
            final_coin = self.eur
        elif final_coin == "try":
            final_coin = self.lira
        elif final_coin == "gbp":
            final_coin = self.gbp
        else:
            print("Wrong coin")
            print("Closing program...")
            sys.exit()
        print (final_coin)

        total = (total_dollars * final_coin) 

        print(f"el total es de:{total} ")

        
menu()

File: Ejercicio2/ejercicio2.js
// 2. Create a currency converter between CLP, ARS, USD, EUR, TRY, GBP with the following features:
// 		The user must choose their initial currency and the currency they want to exchange to.
// 		The user can choose whether or not to withdraw their funds. If they choose not to withdraw, it should return to the main menu.
// 		If the user decides to withdraw the funds, the system will charge a 1% commission.
// 		Set a minimum and maximum amount for each currency, it can be of your choice.
// 		The system should ask the user if they want to perform another operation. If they choose to do so, it should restart the process; otherwise, the system should close.
import inquirer from "inquirer";
const minAmount = 10;
const maxAmount = 9000;
let originCurrency;
let Destinationcurrency;
let amount;
const conversiones = {
  USD: 1,
  EUR: 0.91,
  ARS: 808.47,
  CLP: 888.94,
  TRY: 29.81,
  GBP: 0.79,
};
function convert(originCurrency, Destinationcurrency, amount) {
  const originToUsd = (amount / conversiones[originCurrency]);
  const converteValue = (originToUsd * conversiones[Destinationcurrency]).toFixed(3);
  const commission = (getCommission(converteValue)).toFixed(3);
  const finalValue = (converteValue - commission).toFixed(3);
  return { converteValue, commission, finalValue };
}
function getCommission(amount) {
  return amount * 0.01;
}
const opcionesIniciales = [
  { name: "USD" },
  { name: "EUR" },
  { name: "ARS" },
  { name: "CLP" },
  { name: "TRY" },
  { name: "GBP" },
];

const preguntaMonedaOrigen = {
  type: "list",
  name: "originCurrency",
  message: "Selecciona la moneda de origen:",
  choices: opcionesIniciales,
};
const preguntaMonedaDestino = (originCurrency) => ({
  type: "list",
  name: "destinationcurrency",
  message: "Selecciona la moneda de destino:",
  choices: opcionesIniciales.filter((opcion) => opcion.name != originCurrency),
});
const preguntaCantidad = {
  type: "input",
  name: "amount",
  message: "Ingrese la cantidad de dinero:",
  validate: function (input) {
    const value = parseFloat(input);
    if (isNaN(value) || value <= 0) {
      return "Ingrese un número válido y mayor que cero.";
    } else if (value < minAmount || value > maxAmount) {
      return `Ingrese un valor entre ${minAmount} y ${maxAmount}.`;
    }
    return true;
  },
};

const preguntaConfirmacionTransaccion = {
  type: "confirm",
  name: "decision",
  message: "¿Estás seguro de realizar la transacción?",
  default: false,
};
const preguntarContinuacion = {
  type: "confirm",
  name: "decision",
  message: "¿Deseas realizar otra transaccion?",
  default: false,
};
function iniciarTransaccion() {
  inquirer
  .prompt(preguntaMonedaOrigen)
  .then((respuesta) => {
    originCurrency = respuesta.originCurrency;
    return inquirer.prompt(preguntaMonedaDestino(originCurrency));
  })
  .then((respuesta) => {
    Destinationcurrency = respuesta.destinationcurrency;
    return inquirer.prompt(preguntaCantidad);
  })
  .then((respuesta) => {
    amount = respuesta.amount;
    if (amount < maxAmount && amount < maxAmount) {
    }
    return convert(originCurrency, Destinationcurrency, amount);
  })
  .then((respuesta) => {
    console.log("Cuota por transacción del 1%");
    console.log(`Monto inicial: ${amount} ${originCurrency}`);
    console.log(`Monto después de la conversión: ${respuesta.converteValue} ${Destinationcurrency}`);
    console.log(`Monto final después de la comisión: ${respuesta.finalValue} ${Destinationcurrency}`);
    return inquirer.prompt(preguntaConfirmacionTransaccion);
  })
  .then((respuesta) => {
    if (respuesta.decision) {
      console.log("Transacción realizada con éxito.");
      return inquirer.prompt(preguntarContinuacion);
      
    }else{
        return iniciarTransaccion() 

    }
  })
  .then((respuesta) => {
    if (respuesta && respuesta.decision) {
      iniciarTransaccion() 
    }else{
      console.log("Gracias por usar nuestro servicio. ¡Hasta luego!");
    }
  })
  .catch((error) => {
    console.error("Error:", error);
  });
  
}
console.log("¡Bienvenido a la Aplicación de Conversión de Monedas!");
console.log("--------------------------------------------------");
iniciarTransaccion()

